src/nomad_simulations/physical_property.py

```diff
@@ -0,0 +1,231 @@
+from nomad.metainfo import (
+    Quantity,
+    SubSection,
+    SectionProxy,
+    Reference,
+    Section,
+    Context,
+    MEnum,
+)
+from nomad.metainfo.metainfo import DirectQuantity, Dimension, _placeholder_quantity
+from nomad.datamodel.metainfo.basesections import Entity
+
+from .variables import Variables
+from .numerical_settings import SelfConsistency
+
+
+# We add `logger` for the `PhysicalProperty.variables_shape` method
+logger = utils.get_logger(__name__)
+
+
+class PhysicalProperty(ArchiveSection):
+    """
+    A base section used to define the physical properties obtained in a simulation, experiment, or in a post-processing
+    analysis. Besides its `value`, it also contains:
+        - Quantities that are string identifiers: `source`, `type`, `label`
+        - Quantities for calculating the shape of the physical property: `shape`, `variables`
+        - Quantities for referencing and establishing the character of a physical property: `outputs_ref`, `entity_ref`, `self_consistent_ref`,
+        `is_derived`, `is_scf_converged`
+    """
+
+    name = Quantity(
+        type=str,
+        description="""
+        Name of the physical property. Example: `'ElectronicBandGap'`.
+        """,
+    )
+
+    source = Quantity(
+        type=MEnum('simulation', 'measurement', 'analysis'),
+        default='simulation',
+        description="""
+        Source of the physical property. Example: an `ElectronicBandGap` can be obtained from a `'simulation'`
+        or in an `'measurement'`.
+        """,
+    )
+
+    type = Quantity(
+        type=str,
+        description="""
+        Type categorization of the physical property. Example: an `ElectronicBandGap` can be `'direct'`
+        or `'indirect'`.
+        """,
+        # ! add more examples in the description
+    )
+
+    label = Quantity(
+        type=str,
+        description="""
+        Label for additional classification of the physical property. Example: an `ElectronicBandGap`
+        can be labeled as `'DFT'` or `'GW'` depending on the methodology used to calculate it.
+        """,
+        # ! add more examples in the description
+    )
+
+    rank = DirectQuantity(
+        type=Dimension,
+        shape=['0..*'],
+        default=[],
+        name='rank',
+        description="""
+        Rank of the tensor describing the physical property. This quantity is stored as a Dimension:
+            - scalars (tensor rank 0) have `rank=[]` (`len(rank) = 0`),
+            - vectors (tensor rank 1) have `rank=[a]` (`len(rank) = 1`),
+            - matrices (tensor rank 2), have `rank=[a, b]` (`len(rank) = 2`),
+            - etc.
+        """,
+    )
+
+    variables = SubSection(sub_section=Variables.m_def, repeats=True)
+
+    value: Quantity = _placeholder_quantity
+
+    entity_ref = Quantity(
+        type=Entity,
+        description="""
+        Reference to the entity that the physical property refers to.
+        """,
+    )
```

Comment:
This is again quite vague, but here I think you might have a shot at concretizing.
Within this plugin, you use `Entity` for defining a `GeometricSpace`. Are there any use cases that you're expecting?
I thought that this was rather meant to refer to the proper part in `model_system`.

Reply:
I can put examples in the description, thanks for pointing the vagueness of this.

For you to know, this can ref `AtomsState`, `OrbitalsState`, in order to differentiate `values_per_atom` or `values_per_orbital`. I think we can leave it as abstract as ref to an `Entity` so that it works in, effectively, any case. The ref to `ModelSystem` holds a special situation because of the time evolution (we need a lists of Outputs with refs to the elements in the list of ModelSystems).